service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -165,7 +165,25 @@ func (r *Resource) getMaxCPU(ctx context.Context) (*resource.Quantity, error) {
 		return nil, microerror.Mask(nodeCpuNotFoundError)
 	}
 
-	q, err := quantityMultiply(nodeCpu, 0.9)
+	n, ok := nodeCpu.AsInt64()
+	if !ok {
+		return nil, microerror.Maskf(quantityConversionError, nodeCpu.String())
+	}
+
+	var q *resource.Quantity
+	// If allocatable cpu is less then 6
+	// maxAllowedCPU is 70%
+	if n <= 6 {
+		q, err = quantityMultiply(nodeCpu, 0.7)
+		// If allocatable cpu is more then 12
+		// maxAllowedCPU is 60%
+	} else if n >= 12 {
+		q, err = quantityMultiply(nodeCpu, 0.6)
+		// Default 90%
+	} else {
+		q, err = quantityMultiply(nodeCpu, 0.9)
+	}
```

Comment:
1. let's set the comments inside each "if" so it's easier to spot what it talks about.
2. set the conditions in an ascending order, so it's easier to understand the logic

```suggestion
	if n <= 6 {
		// If allocatable cpu is less than 6
		// maxAllowedCPU is 70% of total
		q, err = quantityMultiply(nodeCpu, 0.7)
	} else if n < 12 {
		// Between 6 and 12 CPUs
		// maxAllowedCPU is 90% of total
		q, err = quantityMultiply(nodeCpu, 0.9)
	} else {
		// More than 12 CPUs
		// maxAllowedCPU is 60% of total
		q, err = quantityMultiply(nodeCpu, 0.6)
	}
```

I don't understand why the bump in %age between 6 and 12 CPUs.
Maybe this code will require a link to the page with your findings regarding CPU usage.